Simonova lesson3

NatashaAravia opened the pull request.

NatashaAravia committed:
homework 3 part

NatashaAravia committed:
homework 3 part 2

ParmaAngularCourse commented on angularHomework/src/app/news-list/news-modal-content/news-modal-content.component.html:
```diff
@@ -0,0 +1,28 @@
+<span class="title">{{news && news.id != 0 ? "Изменить новость" : "Добавить новость"}}</span>
```
Чтобы подсократить шаблон мы можем здесь сделать расчет свойства в классе. И использовать это свойство уже рассчитанное тут. Данные у нас меняются только на @Input и он один, поэтому можно сделать так:

@Input()
  get news(): News | undefined { return this._news; }
  set news(_news: News | undefined) {
    this._news = _news;
    this.title = this._news && this._news.id != 0 ? "Изменить новость" : "Добавить новость";
  }
  public title: string = '';
  private _news: News | undefined;

Либо, если инпутов несколько, и они зависят друг от друга, то можно ставить расчет на хук ngOnChanges()

ParmaAngularCourse commented on angularHomework/src/app/news-list/news-modal-content/news-modal-content.component.html:
```diff
@@ -0,0 +1,28 @@
+    <ng-container *ngIf="showRadio">
+        <div *ngFor="let item of newsTypes">
+            <input type='radio' name="name" [id]="'newsType' + item.id"
+                [value]="item.id" (change)="onChangeInput($event, name)" [checked]="news ? item.id === news.newsType.id : false">
```
Мне кажется, тут логичнее было бы через value тоже провести. Такая внезапная завязка на news только в радио-группе получилась

ParmaAngularCourse commented on angularHomework/src/app/news-list/context-menu/context-menu.component.ts:
```diff
@@ -0,0 +1,34 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { Position } from '../../model/news-type';
+
+@Component({
```
Единственный компонент у нас без OnPush. Может, включим?

ParmaAngularCourse commented on angularHomework/src/app/news-list/context-menu/context-menu.component.ts:
```diff
@@ -0,0 +1,34 @@
+  templateUrl: './context-menu.component.html',
+  styleUrls: ['./context-menu.component.css']
+})
+export class ContextMenuComponent implements OnInit {
```
Тоже, мне кажется, можно в папку shared вынести компонент, он у нас от бизнес-логики не зависит вроде как. Не сильно универсальный конечно, но и про новости в нем ничего нет

NatashaAravia committed:
исправление замечаний + удаление id новости из массива отмеченных при удалении этой новости
